contracts/rewarders/BarRewarderPerSec.sol

```diff
@@ -0,0 +1,119 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.6.12;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/utils/ReentrancyGuard.sol";
+import "../boringcrypto/BoringOwnable.sol";
+import "../libraries/SafeERC20.sol";
+
+interface IBarV2 {
+    function totalSupply() external view returns (uint256);
+
+    function balanceOf(address account) external view returns (uint256);
+
+    function entryFee() external view returns (uint256);
+}
+
+/**
+ * This is a sample contract to be used in the JoeBarV2 to reward moJoe holder with JOE.
+ *
+ * It assumes no minting rights, so requires a set amount of YOUR_TOKEN to be transferred to this contract prior.
+ * E.g. say you've allocated 100,000 XYZ to the JOE-XYZ farm over 30 days. Then you would need to transfer
+ * 100,000 XYZ and set the block reward accordingly so it's fully distributed after 30 days.
+ *
+ */
+contract BarRewarderPerSec is BoringOwnable, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    IERC20 public immutable joe;
+    IBarV2 public immutable bar;
+    uint256 public apr;
+
+    uint256 private secondsPerYear = 365 days;
+
+    uint256 public lastRewardTimestamp;
+    uint256 public tokenPerSec;
+
+    uint256 public unpaidRewards;
+
+    event ClaimReward(uint256 amount);
+    event AprUpdated(uint256 oldRate, uint256 newRate);
+
+    modifier onlyBar() {
+        require(msg.sender == address(bar), "onlyBar: only JoeBar can call this function");
+        _;
+    }
+
+    constructor(
+        IERC20 _joe,
+        IBarV2 _bar,
+        uint256 _apr
+    ) public {
+        require(Address.isContract(address(_joe)), "constructor: reward token must be a valid contract");
+        require(Address.isContract(address(_bar)), "constructor: LP token must be a valid contract");
+
```

Comment:
Add a require statement to set upper bound on APR

Reply:
What do we want it to be ? 100% ?

Reply:
Yes